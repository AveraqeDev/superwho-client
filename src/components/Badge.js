import React from 'react';

import '../styles/Badge.css';

const Badge = ({ values }) => {
  return (
    <>
      {values.split(',').map((value, idx) => {
        return (
          <span key={idx} className="badge">
            {value}
          </span>
        );
      })}
    </>
  )
};

export default Badge;