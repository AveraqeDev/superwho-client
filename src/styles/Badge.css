.badge {
  display: inline-block;
  background: #2A9BE4;
  padding: .4rem;
  margin: .3rem .3rem .3rem 0;
  border-radius: 5px;
  color: #ffffff;
  font-weight: bold;
}