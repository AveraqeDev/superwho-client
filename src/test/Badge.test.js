import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Badge from '../components/Badge';

describe('<Badge />', () => {
  const testData = [
    'Test1',
    'Test2',
    'Test3',
    'Test4',
    'Test5'
  ];

  // Smoke test
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Badge values={testData} />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  //Snapshot test
  it('renders the UI as expected', () => {
    const tree = renderer
      .create(
        <Badge values={testData} />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  })
});