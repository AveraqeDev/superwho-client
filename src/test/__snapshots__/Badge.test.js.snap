// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Badge /> renders the UI as expected 1`] = `
Array [
  <span
    className="badge"
  >
    Test1
  </span>,
  <span
    className="badge"
  >
    Test2
  </span>,
  <span
    className="badge"
  >
    Test3
  </span>,
  <span
    className="badge"
  >
    Test4
  </span>,
  <span
    className="badge"
  >
    Test5
  </span>,
]
`;
